### PR TITLE
Statically linking in `libpcap`, feature-gated

### PR DESCRIPTION
**Pull request by ilya-epifanov**

### CODE REVIEW

File: pcap-sys/build.rs [hunk 1]
```diff
@@ -18,7 +21,31 @@ fn main() {
     println!("cargo:rustc-link-lib=wpcap");
 }
 
-#[cfg(unix)]
+#[cfg(all(unix, feature = "static-libpcap"))]
+fn main() {
+    let additional_includes = std::env::var("ADDITIONAL_INCLUDES").ok();
```
Comment (ilya-epifanov) [inline pcap-sys/build.rs]: need a better name for the env var

File: src/core.rs [hunk 2]
```diff
@@ -208,7 +208,7 @@ impl ErrBuf {
     }
 
     fn as_raw_ptr(&mut self) -> *mut libc::c_char {
-        self.buf.as_mut_ptr()
+        unsafe { (self.buf.as_mut_ptr() as *mut libc::c_char) }
```
Comment (ilya-epifanov) [inline src/core.rs:211]: will move to another PR

File: src/core.rs [hunk 3]
```diff
@@ -359,6 +359,12 @@ impl Handle {
             println!("error");
         }
     }
+
+    pub fn set_immediate_mode(&self, immediate: bool) {
```
Comment (ilya-epifanov) [inline src/core.rs:363]: will move to another PR

**Comment by jmmk** [comment]
@ilya-epifanov I'm not actively using this project (or even Rust language). Would you be interested in taking over ownership of this project? If not, I can at least give you push access and you can feel free to add these changes

**Comment by ilya-epifanov** [comment]
@jmmk Hi there!

Actually, it's not a bad idea, since I'm gonna use this crate for a production project.

**Comment by jmmk** [comment]
@ilya-epifanov It looks like I can't transfer because you already have a repo `rustcap` (your fork). I've invited you to be a contributor which should give you push access, and we can try the transfer again later

I've also invited you to be an owner of the crates on `crates.io`

**Comment by sekineh** [comment]
Hi I recommend you also add build command to CI scripts to test the added feature.

https://github.com/jmmk/rustcap/blob/1684f3c23e98660c62cff8958bb1c52b5aa85985/.travis.yml#L20
https://github.com/jmmk/rustcap/blob/1684f3c23e98660c62cff8958bb1c52b5aa85985/appveyor.yml#L18
